### PR TITLE
Support transactions

### PR DESCRIPTION
**Pull request by YordanPavlov**
A pull request to expose the transactions support of LibrdKafka.

This PR is a copy of the PR:
https://github.com/Blizzard/node-rdkafka/pull/881

some commits were cleaned by interactive rebase.

### CODE REVIEW

**Comment by iradul** [comment]
I pushed the code for async version. We now need to make proper e2e tests.

File: lib/producer.js [hunk 1]
```diff
@@ -295,16 +294,18 @@ Producer.prototype.disconnect = function(timeout, cb) {
...
+ * @param {number} timeout - Number of milliseconds to try to initialize before giving up, defaults to 5 seconds.
+ * @param {function} cb - Callback to return when operation is completed
  * @return {Producer} - returns itself.
  */
- Producer.prototype.initTransactions = function(timeout) {
-  if (timeout === undefined || timeout === null) {
+Producer.prototype.initTransactions = function(timeout, cb) {
+  if (typeof timeout === 'function') {
     timeout = 5000;
+    cb = timeout;
```
Comment (thynson) [inline lib/producer.js:304]: This shall be done before assign before `timeout=5000`

File: lib/producer.js [hunk 2]
```diff
@@ -314,48 +315,61 @@ Producer.prototype.disconnect = function(timeout, cb) {
...
+ * @param {number} timeout - Number of milliseconds to try to commit before giving up, defaults to 5 seconds
+ * @param {function} cb - Callback to return when operation is completed
  * @return {Producer} - returns itself.
  */
- Producer.prototype.commitTransaction = function(timeout) {
-  if (timeout === undefined || timeout === null) {
+Producer.prototype.commitTransaction = function(timeout, cb) {
+  if (typeof timeout === 'function') {
     timeout = 5000;
+    cb = timeout;
```
Comment (thynson) [inline lib/producer.js:334]: Same as above

File: lib/producer.js [hunk 3]
```diff
@@ -314,48 +315,61 @@ Producer.prototype.disconnect = function(timeout, cb) {
...
+ * @param {number} timeout - Number of milliseconds to try to abort, defaults to 5 seconds
+ * @param {function} cb - Callback to return when operation is completed
  * @return {Producer} - returns itself.
  */
- Producer.prototype.abortTransaction = function(timeout) {
-  if (timeout === undefined || timeout === null) {
+Producer.prototype.abortTransaction = function(timeout, cb) {
+  if (typeof timeout === 'function') {
     timeout = 5000;
+    cb = timeout;
```
Comment (thynson) [inline lib/producer.js:351]: Same bug as above.

**Comment by thynson** [comment]
I tried this feature locally, except TypeError caused by adding async support, all the other parts are good to me so far.

File: lib/producer.js (hunk 1 above)
Comment (iradul) [reply lib/producer.js:304]: Good catch

**Comment by iradul** [comment]
The only missing thing is updated docs, but this can be done in a separate PR or commit. Thank you @YordanPavlov, and everyone who participated.

**Pull request merged**